Userland/Applications/PixelPaint/Layer.cpp

```diff
@@ -180,6 +180,31 @@ void Layer::crop(Gfx::IntRect const& rect)
     did_modify_bitmap();
 }
 
+void Layer::resize(Gfx::IntSize const& new_size, Gfx::Painter::ScalingMode scaling_mode)
+{
+    const Gfx::IntRect old_rect(Gfx::IntPoint(0, 0), size());
+    const Gfx::IntRect new_rect(Gfx::IntPoint(0, 0), new_size);
+
+    {
+        auto resized = Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRA8888, new_size).release_value_but_fixme_should_propagate_errors();
+        Gfx::Painter painter(resized);
```

Comment:
Do we really need to bother the painter here? The `did_change_rect` up on the Image::resize should take care of repainting already.